File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
    let answer = []
    let students = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    let scores = [0, 0, 0]
    
    for (let i=0; i<students.length; i++){
        let rlt = 0
        
        for (let j=0; j<answers.length; j++){
            tmp = students[i][j % students[i].length]
            
            if (answers[j] === tmp) {
                rlt++
            }
        }
        scores[i] = rlt
    }
    
    for (let i=0; i<scores.length; i++){
        if (scores[i] === Math.max(...scores)){
            answer.push(i+1)
        }
    }
    return answer
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    students = [[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    scores = [0, 0, 0]
    
    for i in range(len(students)):
        for j in range(len(answers)):
            if answers[j] == students[i][j % (len(students[i]))]:
                scores[i] += 1
    
    for i, k in enumerate(scores):
        if k == max(scores):
            answer.append(i+1)
    
    return answer